tests/integration/suite/actors/healthz/endpoint/noappentities.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implien.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoint
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(noappentities))
+}
+
+// noappentities ensures that actors work even though no entities have been
+// defined and the app healthz endpoint is disabled.
+type noappentities struct {
+	daprdNoHealthz  *daprd.Daprd
+	daprd           *daprd.Daprd
+	place           *placement.Placement
+	healthzCalled   chan struct{}
+	noHealthzCalled chan struct{}
+}
+
+func (n *noappentities) Setup(t *testing.T) []framework.Option {
+	n.healthzCalled = make(chan struct{})
+	n.noHealthzCalled = make(chan struct{})
+	n.place = placement.New(t)
+
+	var onceHealthz, onceNoHealthz sync.Once
+
+	srvHealthz := prochttp.New(t,
+		prochttp.WithHandlerFunc("/dapr/config", func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte(`{"entities": ["myactortype"]}`))
+		}),
+		prochttp.WithHandlerFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			onceHealthz.Do(func() {
+				close(n.healthzCalled)
+			})
+		}),
+		prochttp.WithHandlerFunc(pathMethodFoo, func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte(`OK`))
+		}),
+	)
+
+	srvNoHealthz := prochttp.New(t,
+		prochttp.WithHandlerFunc("/dapr/config", func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte(`{}`))
+		}),
+		prochttp.WithHandlerFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+			onceNoHealthz.Do(func() {
+				close(n.noHealthzCalled)
+			})
+		}),
+	)
```

Comment:
qq when there are no actor types and health check is disabled, why is a healthz endpoint needed and is being called?